### ISSUE TITLE
console-backend pod crashing with OOM

### ISSUE THREAD

**Issue by sbgreene1307** [kyma-project/kyma#6667]
**Description**
 
Tech. Details:
AKS
Kyma 1.8
Kubernetes 1.14.8

We've created 3 new AKS clusters and installed Kyma 1.8. Two of the three clusters have issues with core-console-backend-service pod running into OOM errors. Not sure how to reliably reproduce this. 

The pod eventually came up after 64 restarts.

Logs from the previous container in the pod
```
I1210 23:50:57.374753       1 module.go:19] Making the 'assetstore' module pluggable...
I1210 23:50:57.375359       1 module.go:19] Making the 'cms' module pluggable...
I1210 23:50:57.375552       1 module.go:19] Making the 'servicecatalog' module pluggable...
I1210 23:50:57.375648       1 module.go:19] Making the 'servicecatalogaddons' module pluggable...
I1210 23:50:57.376814       1 module.go:19] Making the 'application' module pluggable...
I1210 23:50:57.379063       1 module.go:19] Making the 'apicontroller' module pluggable...
I1210 23:50:57.379089       1 module.go:19] Making the 'apigateway' module pluggable...
I1210 23:50:57.379316       1 module.go:19] Making the 'authentication' module pluggable...
I1210 23:50:57.379658       1 module.go:19] Making the 'serverless' module pluggable...
I1210 23:50:58.493948       1 eventhandler.go:62] Enabling module assetstore...
I1210 23:50:58.493984       1 eventhandler.go:62] Enabling module authentication...
I1210 23:50:58.494814       1 eventhandler.go:62] Enabling module application...
I1210 23:50:58.495586       1 eventhandler.go:62] Enabling module servicecatalog...
I1210 23:50:58.495853       1 eventhandler.go:62] Enabling module servicecatalogaddons...
I1210 23:50:58.495950       1 eventhandler.go:62] Enabling module cms...
I1210 23:50:58.496552       1 eventhandler.go:62] Enabling module apigateway...
I1210 23:50:58.498371       1 eventhandler.go:62] Enabling module apicontroller...
I1210 23:50:58.789741       1 main.go:168] Listening on 0.0.0.0:3000
E1210 23:51:05.248682       1 middleware.go:39] Unable to authenticate the request due to an error: oidc: authenticator not initialized
```

**Expected result**

core-console-backend doesn't run out of memory

**Actual result**

core-console-backend goes into CrashloopBackoff because of OOM termination

**Steps to reproduce**

**Troubleshooting**

**Comment by kwiatekus** [comment]
Hi @sbgreene1307 
Could  you share  installation status for of your runtime? (`installer.yaml`)
`kubectl get installations.installer.kyma-project.io -oyaml > installer.yaml`.

**Comment by sbgreene1307** [comment]
```
apiVersion: installer.kyma-project.io/v1alpha1
kind: Installation
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"installer.kyma-project.io/v1alpha1","kind":"Installation","metadata":{"annotations":{},"finalizers":["finalizer.installer.kyma-project.io"],"labels":{"action":"install","kyma-project.io/installation":""},"name":"kyma-installation","namespace":"default"},"spec":{"components":[{"name":"cluster-essentials","namespace":"kyma-system"},{"name":"testing","namespace":"kyma-system"},{"name":"istio-init","namespace":"istio-system"},{"name":"istio","namespace":"istio-system"},{"name":"xip-patch","namespace":"kyma-installer"},{"name":"istio-kyma-patch","namespace":"istio-system"},{"name":"knative-serving-init","namespace":"knative-serving"},{"name":"knative-serving","namespace":"knative-serving"},{"name":"knative-eventing","namespace":"knative-eventing"},{"name":"prometheus-operator","namespace":"kyma-system"},{"name":"dex","namespace":"kyma-system"},{"name":"ory","namespace":"kyma-system"},{"name":"api-gateway","namespace":"kyma-system"},{"name":"service-catalog","namespace":"kyma-system"},{"name":"service-catalog-addons","namespace":"kyma-system"},{"name":"helm-broker","namespace":"kyma-system"},{"name":"nats-streaming","namespace":"natss"},{"name":"assetstore","namespace":"kyma-system"},{"name":"cms","namespace":"kyma-system"},{"name":"core","namespace":"kyma-system"},{"name":"knative-provisioner-natss","namespace":"knative-eventing"},{"name":"event-bus","namespace":"kyma-system"},{"name":"application-connector-ingress","namespace":"kyma-system"},{"name":"application-connector-helper","namespace":"kyma-integration"},{"name":"application-connector","namespace":"kyma-integration"},{"name":"backup-init","namespace":"kyma-system"},{"name":"backup","namespace":"kyma-system"},{"name":"logging","namespace":"kyma-system"},{"name":"jaeger","namespace":"kyma-system"},{"name":"monitoring","namespace":"kyma-system"},{"name":"kiali","namespace":"kyma-system"}],"url":"","version":"0.0.1"}}
  finalizers:
  - finalizer.installer.kyma-project.io
  labels:
    kyma-project.io/installation: ""
  name: kyma-installation
  namespace: default
spec:
  components:
  - name: cluster-essentials
    namespace: kyma-system
    release: ""
  - name: testing
    namespace: kyma-system
    release: ""
  - name: istio-init
    namespace: istio-system
    release: ""
  - name: istio
    namespace: istio-system
    release: ""
  - name: xip-patch
    namespace: kyma-installer
    release: ""
  - name: istio-kyma-patch
    namespace: istio-system
    release: ""
  - name: knative-serving-init
    namespace: knative-serving
    release: ""
  - name: knative-serving
    namespace: knative-serving
    release: ""
  - name: knative-eventing
    namespace: knative-eventing
    release: ""
  - name: prometheus-operator
    namespace: kyma-system
    release: ""
  - name: dex
    namespace: kyma-system
    release: ""
  - name: ory
    namespace: kyma-system
    release: ""
  - name: api-gateway
    namespace: kyma-system
    release: ""
  - name: service-catalog
    namespace: kyma-system
    release: ""
  - name: service-catalog-addons
    namespace: kyma-system
    release: ""
  - name: helm-broker
    namespace: kyma-system
    release: ""
  - name: nats-streaming
    namespace: natss
    release: ""
  - name: assetstore
    namespace: kyma-system
    release: ""
  - name: cms
    namespace: kyma-system
    release: ""
  - name: core
    namespace: kyma-system
    release: ""
  - name: knative-provisioner-natss
    namespace: knative-eventing
    release: ""
  - name: event-bus
    namespace: kyma-system
    release: ""
  - name: application-connector-ingress
    namespace: kyma-system
    release: ""
  - name: application-connector-helper
    namespace: kyma-integration
    release: ""
  - name: application-connector
    namespace: kyma-integration
    release: ""
  - name: backup-init
    namespace: kyma-system
    release: ""
  - name: backup
    namespace: kyma-system
    release: ""
  - name: logging
    namespace: kyma-system
    release: ""
  - name: jaeger
    namespace: kyma-system
    release: ""
  - name: monitoring
    namespace: kyma-system
    release: ""
  - name: kiali
    namespace: kyma-system
    release: ""
  url: ""
  version: 0.0.1
status:
  conditions:
  - lastProbeTime: "2019-12-09T22:29:11Z"
    lastTransitionTime: "2019-12-09T22:29:11Z"
    status: "True"
    type: Installed
  - lastProbeTime: "2019-12-09T22:29:11Z"
    lastTransitionTime: "2019-12-09T22:29:11Z"
    status: "False"
    type: Installing
  - lastProbeTime: "2019-12-09T22:29:11Z"
    lastTransitionTime: "2019-12-09T22:29:11Z"
    status: "False"
    type: InProgress
  - lastProbeTime: "2019-12-09T22:29:11Z"
    lastTransitionTime: "2019-12-09T21:54:29Z"
    status: "False"
    type: Error
  - lastProbeTime: "2019-12-09T22:29:11Z"
    lastTransitionTime: "2019-12-09T22:29:11Z"
    status: "False"
    type: Uninstalled
  description: Kyma installed
  errorLog: []
  state: Installed
  url: ""
  version: 0.0.1
```

**Comment by parostatkiem** [comment]
Hi @sbgreene1307, I think we found the solution. It's in  _beta_ phase right now but has already proven to work on various clusters under a huge load. I think you can try it on your cluster.
The only thing you need to do is to set `requests.memory` to `96Mi` in the _core-console-backend-service_ deployment. You can do this either in UI by editing the mentioned deployment this way:
![image](https://user-images.githubusercontent.com/22825433/72250339-a8022e00-35fb-11ea-8908-b2c7f1239a9b.png)
or similarly from terminal by 
```bash
kubectl edit deployment -n kyma-system core-console-backend-service 
```

If this solution proves to be good for some longer time, we'll include it in a release.
Anyway, please let us know if it helped you :)

**Comment by parostatkiem** [comment]
@sbgreene1307 Update:
We the previous solution didn't seem to work, however we found another one that is proven to make things better. Sadly, it doesn't solve the problem completely but makes it happening much less often.
It'll be merged to _master_ soon and probably (@kwiatekus please confirm or deny) will be included in the next release.
If you're interested, I'm talking about #6887

**Comment by kwiatekus** [comment]
We have identified the root causes of the OOM problems of console-backend-service (CBS):

* Most of the time CBS keeps it memory usage below 64 MB. All the queries utilise its internal cache which is memory efficient. We realise that it is enough to set [requested memory](https://github.com/kyma-project/kyma/pull/6904/files#diff-58c94eeebeed6b0078bbf7b8dcae7ce4R43) at the level of `64 Mi`
* **There are peaks of memory consumption**. They occur when CBS is synchronising its cache with API server as  involves JSON objects parsing (memory expensive). They happen at the very beginning when CBS is started and periodically, according to the re-sync policy (every 10 minutes).
Peak values often exceeded the limit, therefore we **[increased the limit](https://github.com/kyma-project/kyma/pull/6904/files#diff-58c94eeebeed6b0078bbf7b8dcae7ce4R41)** to `300 Mi`
* CBS is divided into several domains and each domain use its own http client to sync data with API  server. We adopted the cache re-sync policy so that re-sync operations for different domains are not happening in the same time - therefore, **higher memory consumption at the time of re-sync is more distributed**. Additionally, we have limited the default API burst rate - limiting the number of simultaneous objects to parse. (https://github.com/kyma-project/kyma/pull/6887)
* We have identified and **fixed memory leak** - unclosed connection upon authentication error (https://github.com/kyma-project/kyma/pull/6870)
